ramalama/common.py

```diff
@@ -55,14 +57,45 @@ def container_manager():
     return _engine
 
 
-def is_podman_machine_running_with_krunkit():
-    podman_machine_list = ["podman", "machine", "list", "--all-providers"]
+def apple_vm():
+    podman_machine_list = ["podman", "machine", "list", "--format", "json", "--all-providers"]
     try:
-        output = run_cmd(podman_machine_list, ignore_stderr=True).stdout.decode("utf-8").strip()
-        return re.search("krun.*running", output)
+        # Run the command and decode the output
+        machines_json = run_cmd(podman_machine_list, ignore_stderr=True).stdout.decode("utf-8").strip()
+
+        # Parse the JSON into Python objects
+        machines = json.loads(machines_json)
+
+        for machine in machines:
+            name = machine.get("Name")
+            provider = machine.get("VMType")
+            running = machine.get("Running")
+
+            if provider == "applehv" and running:
+                # Ask the user if they want to use the "applehv" provider
+                while True:
+                    user_input = (
+                        input(
+                            f"Warning! Your VM {name} is using {provider}, which does not support GPU. "
+                            "Only the provider libkrun has GPU support. "
+                            "Do you want to proceed without GPU? (yes/no): "
+                        )
+                        .strip()
+                        .lower()
+                    )  # Make sure input is stripped and lowercase
+
+                    if user_input in ["yes", "y"]:
+                        os.environ["LIBKRUN"] = "False"
```

Comment:
We might as well delete these:

os.environ["LIBKRUN"]

sets, they aren't used anywhere.

FWIW, I do try to avoid using environment variables. Being global variables, it's preferable to avoid. I did add usage of environment variables like HIP_VISIBLE_DEVICES, CUDA_VISIBLE_DEVICES but these environment variables pre-date RamaLama, that's trying to take advantage of existing functionality and not re-invent the wheel.

Can be caught in a follow on PR though.

Reply:
Wait I see what you did now, still not ideal to use an env var. Anyway this is good enough for merge at least.